clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -989,7 +1027,7 @@ private long sendEligibleCalls(long now) {
                     continue;
                 }
                 Call call = calls.remove(0);
-                int timeoutMs = calcTimeoutMsRemainingAsInt(now, call.deadlineMs);
+                int timeoutMs = Math.min(requestTimeoutMs, calcTimeoutMsRemainingAsInt(now, call.deadlineMs));
```

Comment:
In addition to passing through the timeout to `createRequest`, we also need to pass it through in `newClientRequest`. Only some of the APIs support an explicit request timeout. This appears to have been a pre-existing bug.